src/app/pages/rpc-send-transfer/rpc-send-transfer.tsx

```diff
@@ -1,34 +1,232 @@
+  const { broadcastTx } = useBitcoinBroadcastTransaction();
+
   useBreakOnNonCompliantEntity(recipientsAddresses);
 
+  function onCancel() {
+    closeWindow();
+  }
+
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [isBroadcasting, setIsBroadcasting] = useState(false);
+
+  const isInsufficientBalance = btcBalance.balance.availableBalance.amount.isLessThan(
+    amountAsMoney.amount
+  );
+
+  async function onApprove() {
+    setIsBroadcasting(true);
+
+    function onError(e: unknown) {
+      setIsBroadcasting(false);
+      logger.error('Error broadcasting tx', e);
+      navigate(RouteUrls.SendBtcError, {
+        state: {
+          error: e,
+        },
+      });
+    }
+
+    try {
+      const feeRate = selectedFeeData.feeRate;
+      const resp = await generateTx({ amount: amountAsMoney, recipients }, feeRate, utxos);
+
+      if (!resp) return logger.error('Attempted to generate raw tx, but no tx exists');
+
+      const tx = await signTransaction(resp.psbt);
+
+      tx.finalize();
+
+      await broadcastTx({
+        tx: tx.hex,
+        async onSuccess(txid) {
+          setIsBroadcasting(false);
+
+          void analytics.track('broadcast_transaction', {
+            symbol: 'btc',
+            amount: amountAsMoney.amount.toNumber(),
+          });
+
+          chrome.tabs.sendMessage(
+            tabId ?? 0,
+            makeRpcSuccessResponse('sendTransfer', {
+              id: requestId,
+              result: { txid },
+            })
+          );
+
+          setIsSubmitted(true);
+        },
+        onError,
+      });
+    } catch (error) {
+      onError(error);
+    } finally {
+      setIsBroadcasting(false);
+    }
+  }
+
+  const isLoading = btcBalance.isLoadingAllData;
+
+  function getApproverActions() {
+    if (isLoading) {
+      return [
+        <SkeletonLoader key="skeleton" isLoading height="40px" />,
+        <SkeletonLoader key="skeleton" isLoading height="40px" />,
+      ];
+    }
+
+    if (isInsufficientBalance) {
+      return [];
+    }
+
+    if (isBroadcasting) {
+      return [
+        <Button key="submitting" fullWidth aria-busy disabled>
+          Submitting...
+        </Button>,
+      ];
+    }
+
+    if (isSubmitted) {
+      return [
+        <Button key="submitting" fullWidth variant="success" disabled>
+          <HStack justifyContent="center" alignItems="center" gap="space.02">
+            <CheckmarkIcon color="ink.text-primary" variant="small" />
+            <styled.span textStyle="label.02">Submitted</styled.span>
+          </HStack>
+        </Button>,
+      ];
+    }
+
+    return [
+      <Button key="cancel" onClick={onCancel} fullWidth variant="outline">
+        <styled.span textStyle="label.02">Cancel</styled.span>
+      </Button>,
+      <Button type="submit" key="approve" onClick={onApprove} fullWidth>
+        <styled.span textStyle="label.02">Approve</styled.span>
+      </Button>,
+    ];
+  }
```

Comment:
Better a `useMemo` than a fn. This rerenders on every render, vs on dependency change